java/pom.xml

```diff
@@ -688,6 +702,29 @@
        <arrow.vector.classifier>shade-format-flatbuffers</arrow.vector.classifier>
       </properties>
     </profile>
+
+    <!-- using github.com/google/error-prone-javac is required when running on JDK 8 -->
+    <profile>
```

Comment:
i would request this to be the default please :)

Reply:
ditto

Reply:
@tianchen92 
I was asking if anyone building arrow now needs to do so using java9 for this to work? 
(or) can we continue using jdk8.

If we can continue running using jdk8, the change looks good.

If consumers need java9 to build - then i would request running error prone only if a specific profile is used (for e.g. in the example you have given using profile jdk1.8 activates the check with some special javac properties, meaning you would use a command like mvn -P jdk-18 to activate the profile) This way existing consumers will not be forced to upgrade their entire build chain to java9 to use arrow java.

Reply:
I see, thanks for clarify, now it works fine with jdk8 :)

Reply:
awesome..do we also want to modify ci to run this profile, so that it catches further bugs?

Reply:
We don't need do anything, just use 'mvn clean install' will trigger error prone :)

Reply:
sounds good thanks @tianchen92